core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -444,10 +459,52 @@ private void removeImage(String dockerImageName) {
         }
     }
 
-    private void setHook() {
+    private void prune(PruneType pruneType, List<Map.Entry<String, String>> filters) {
+        String[] labels = filters.stream()
+            .filter(it -> "label".equals(it.getKey()))
+            .map(Map.Entry::getValue)
+            .toArray(String[]::new);
+        switch (pruneType) {
+            // Docker only prunes stopped containers, so we have to do it manually
+            case CONTAINERS:
+                List<Container> containers = dockerClient.listContainersCmd()
+                    .withFilter("label", Arrays.asList(labels))
+                    .withShowAll(true)
+                    .exec();
+
+                containers.parallelStream().forEach(container -> {
+                    stopContainer(container.getId(), container.getImage());
```

Comment:
I wonder if the `stopContainer` method name has become misleading and should be renamed. When I read this, my first thought was that this overall switch construct would stop containers but not explicitly remove them.

Reply:
`stop` behaves like this for a long time (since forever?) of course (and differentiates from the semantics of Docker's `container stop`. So while I was always in favor of aligning our domain language to the Docker one, by now this would a such a breaking change of semantics in our API, I am unsure how to easily deliver it?

Reply:
`stopContainer` specifically is a private method, and IIRC is only called by this and a `stopAndRemove` (public) method.

So maybe more do-able?

(+100 for aliging to docker's domain language)

Reply:
Happy to add it to the scope - do you have naming recommendations? `reapContainer`? :D

Reply:
@rnorth Ok you are right, I was mixing it up with our public `stop()` method, which will call `stopAndRemoveContainer` (which I consider confusing but hard to change).  

@bsideup WDTY about just `removeContainer`, since this would be aligned with the methods for other Docker resources?
```java
  public synchronized void performCleanup() {
      registeredContainers.forEach(this::stopContainer);
      registeredNetworks.forEach(this::removeNetwork);
      registeredImages.forEach(this::removeImage);
  }
```

Reply:
sounds good to me 👍